controllers/configurationpolicy_controller.go

```diff
@@ -429,6 +501,66 @@ func (r *ConfigurationPolicyReconciler) handleObjectTemplates(plc policyv1.Confi
 		return
 	}
 
+	// object handling for when configurationPolicy is deleted
+	if plc.Spec.PruneObjectBehavior == "DeleteIfCreated" || plc.Spec.PruneObjectBehavior == "DeleteAll" {
+		// set finalizer if it hasn't been set
+		if len(plc.GetFinalizers()) == 0 {
+			plc.SetFinalizers([]string{"policy.open-cluster-management.io/delete-related-objects"})
+			err := r.Update(context.TODO(), &plc)
+			if err != nil {
+				log.V(1).Error(err, "Error setting finalizer for configuration policy", plc)
+			}
+		}
+
+		// kick off object deletion if configurationPolicy has been deleted
+		if plc.ObjectMeta.DeletionTimestamp != nil {
+			log.V(1).Info("Config policy has been deleted, handling child objects")
+
+			failures := r.cleanUpChildObjects(plc, string(plc.Spec.PruneObjectBehavior))
+
+			if len(failures) == 0 {
+				log.V(1).Info("Objects have been successfully cleaned up, removing finalizer")
+				plc.SetFinalizers([]string{})
+				err := r.Update(context.TODO(), &plc)
+				if err != nil {
+					log.V(1).Error(err, "Error unsetting finalizer for configuration policy", plc)
+				}
+			} else {
+				log.V(1).Info("Object cleanup failed, some objects have not been deleted from the cluster")
+
+				statusChanged := addConditionToStatus(
+					&plc,
+					0,
+					false,
+					"Error cleaning up child objects",
+					"Failed to delete objects: "+strings.Join(failures, ", "))
+				if statusChanged {
+					parentStatusUpdateNeeded = true
+
+					r.Recorder.Event(
+						&plc,
+						eventWarning,
+						fmt.Sprintf(plcFmtStr, plc.GetName()),
+						convertPolicyStatusToString(&plc),
+					)
+				}
+
+				r.checkRelatedAndUpdate(plc, relatedObjects, oldRelated, parentStatusUpdateNeeded)
+			}
+
+			return
+		}
+	} else {
+		// if pruneObjectBehavior is none, no finalizer is needed
+		if len(plc.ObjectMeta.Finalizers) != 0 {
+			plc.SetFinalizers([]string{})
+			err := r.Update(context.TODO(), &plc)
+			if err != nil {
+				log.V(1).Error(err, "Error unsetting finalizer for configuration policy", plc)
+			}
+		}
```

Comment:
This will clobber any finalizers anything else might add to the ConfigurationPolicy, which could be bad.